### PR TITLE
Docs: Added guide on using WSL

### PR DESCRIPTION
**Pull request by agzuniverse**

### CODE REVIEW

File: docs/howto/windows.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+### Accessing the filesystem
+You can access your Windows local drives by navigating to `/mnt/<drive-letter>`
+
+### Running npm commands
+First install node in your subsystem. For example, to install node 8.x in Ubuntu, run:
+
+`curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -`
+`sudo apt-get install -y nodejs`
+
+Install `react-native-cli` globally with `npm i -g react-native-cli`.
```
Comment (borisyankov) [inline docs/howto/windows.md]: This needs to have a `sudo` in front.

File: docs/howto/windows.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+### Modify PATH variable
+Run the following commands so the subsystem checks for npm in `/usr/bin` before it checks `C\Program Files` and tries to run the Windows installation (which of course, does not work in Linux).
+
+`export PATH="$HOME/bin:$HOME/.local/bin:/usr/bin:$PATH"`
+`source ~/.profile`
+
+### Run emulator
```
Comment (borisyankov) [inline docs/howto/windows.md]: Now would be a good place to tell the developer to run `yarn`.
They need to install the dependencies before running the project.

They can also run `npm` that they have installed with `node` but we recommend using `yarn`.

That means a previous step would be to explain how to install `yarn`.

File: docs/howto/windows.md (hunk 1 above)
Comment (agzuniverse) [reply docs/howto/windows.md]: Change made.

File: docs/howto/windows.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+### Install WSL
+Install your preferred distribution of Linux available in the Windows store (You must have Windows build 1607 or later to be able to do this). I used Ubuntu for this guide.
+
+### Accessing the filesystem
+You can access your Windows local drives by navigating to `/mnt/<drive-letter>`
+
+### Running npm commands
+First install node in your subsystem. For example, to install node 8.x in Ubuntu, run:
+
+`curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -`
```
Comment (borisyankov) [inline docs/howto/windows.md]: If you write these like this, they will format better:
```
one line
another line
```

File: docs/howto/windows.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+Small guide on how to use the terminal using WSL (Windows Subsystem for Linux) with the windows file system, and using it to perform any commands while running a react native android app in Android emulator on Windows.
```
[the remark this answers is not in the document]
Comment (agzuniverse) [reply docs/howto/windows.md]: Added a "Why use WSL?" section.

File: docs/howto/windows.md (hunk 3 above)
Comment (agzuniverse) [reply docs/howto/windows.md]: Fixed!

**Comment by borisyankov** [comment]
@gnprice I did run the whole process on a Windows machine that I had intended to install WSL anyway. I confirm that factually the guide is correct.

File: docs/howto/windows.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (borisyankov) [reply docs/howto/windows.md]: I know writing clear and persuasive paragraphs is hard, so not sure how much we want to spend on polishing the intro paragraph (@gnprice opinion?)

I think a single paragraph that combines the first one and `why use` one is enough, as long as it is answering the dev's questions.

A developer has the option of simply running the commands on a Windows machine, so maybe compare Windows terminal vs WSL terminal?

[the remark this answers is not in the document]
Comment (agzuniverse) [reply docs/howto/windows.md]: Well, the Unix terminal is just better (at least it's certainly better out of the box), explaining the differences is probably way out of the scope of our docs?

File: docs/howto/windows.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+Zulip recommends using `yarn` over `npm` which comes with node.
+
+Install yarn with `sudo npm i -g yarn`
+
+### Modify PATH variable
+Run the following commands so the subsystem checks for npm in `/usr/bin` before it checks `C\Program Files` and tries to run the Windows installation (which of course, does not work in Linux).
+
+`export PATH="$HOME/bin:$HOME/.local/bin:/usr/bin:$PATH"`
+`source ~/.profile`
```
Comment (gnprice) [inline docs/howto/windows.md]: I don't really understand these pair of commands! That is, I know what they do -- but not why they're a way to solve this.

What's the contents of your `~/.profile`? Perhaps that would shed some light.

File: docs/howto/windows.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+```
+curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
+sudo apt-get install -y nodejs
+```
+
+Install `react-native-cli` globally with `sudo npm i -g react-native-cli`.
+
+Zulip recommends using `yarn` over `npm` which comes with node.
+
+Install yarn with `sudo npm i -g yarn`
```
Comment (gnprice) [inline docs/howto/windows.md]: We already have instructions for installing Yarn -- they say to go to https://yarnpkg.com/en/docs/install . No need to say something else. :-)

File: docs/howto/windows.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (gnprice) [reply docs/howto/windows.md]: Definitely in favor of clearly articulating the "why". :-)

This version was helpful! I added a bit more in my followup commit d1b7a28cd.

**Comment by gnprice** [comment]
(And thanks @borisyankov for the extremely helpful play-testing that it works!)

**Pull request merged**